### ISSUE TITLE
Failed to build project with xcode 11.4

### ISSUE THREAD

**Issue by bogdangolubcr** [krzysztofzablocki/Sourcery#817]
Hi, when i tried to build my project with new xcode 11.4 I had this issue with no error description.
```
Using configuration file at '.sourcery.platform.yml'
Scanning sources...
	loadOrParse: 0.8185429573059082
	resolution took 0.1482919454574585
	combiningTypes: 0.16124391555786133
	total: 0.9820070266723633
Found 213 types.
Loading templates...
Loaded 6 templates.
	Loading took 0.008983016014099121
Generating code...
error: 
```

**Comment by hartbit** [comment]
Same here.

**Comment by nicholaslythall** [comment]
I was having the same issue. If I ran Sourcery via the terminal it worked fine, but when running as a build phase it failed with `error:`

For some reason, I had the `--disableCache` flag set on the build phase script, and it worked after I removed that, but only if I ran it at least once via CLI before trying to build.

**Comment by nicholaslythall** [comment]
I was able to find a temporary workaround by calling `unset SDKROOT` in the build phase script before calling sourcery.
The value for `SDKROOT` before being unset was `/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk`

**Comment by ilyapuchka** [comment]
Does this happen with any kind of templates or might be related just to Swift templates?

**Comment by RenGate** [comment]
@ilyapuchka In my case this error is only present when swifttemplate is used. What's interesting is that even if the swifttemplate itself is empty, the error is still there. I'm using Sourcery 0.17.0.

**Comment by hartbit** [comment]
For what it's worth, I also have a Swift template.

**Comment by vCrespoP** [comment]
Bumping this issue. 

It's also my case with Sourcery 0.18.0 although it compiles and generates code, but throws an error:

> Command PhaseScriptExecution emitted errors but did not return a nonzero exit code to indicate failure

**Comment by jonnokim17** [comment]
i'm getting the same result as @vCrespoP. do we have a fix for this yet? thanks!

**Comment by krzysztofzablocki** [comment]
@vCrespoP @jonnokim17 does clearing cache change anything?

**Comment by xploit42** [comment]
Same problem here, clearing the cache doesn't change anything

**Comment by mgray88** [comment]
FWIW, I experienced the same issue on 11.4.1, however setting `--disableCache` on the build phase script actually eliminated the error. Subsequently removing it however, caused the error to come back

**Comment by jonnokim17** [comment]
@krzysztofzablocki where exactly would i put the `--disableCache` in the build phase? 

this is what i have so far:
`"$PODS_ROOT"/Sourcery/bin/sourcery --sources "$SRCROOT"/{project_name}/Code --templates "$SRCROOT"/Templates/Sourcery --output "$SRCROOT"/{project_name}/Code/Helpers/Generated`

**Comment by degterev** [comment]
I was having the same problem with swift templates when I run sourcery with brew bundle like 
```bash
brew bundle exec -- sourcery --config Configs/Project/.sourcery.yml --disableCache
```
I'm not sure, but I think the problem is somewhere in env variables because the brew bundle changes some CMAKE paths.

So, the temporary solution for me is just calling sourcery directly.

**Comment by krzysztofzablocki** [comment]
could any of you provide a minimal example code / template that causes the issue for you? I've some time to investigate this now

**Comment by jonnokim17** [comment]
any updates on this yet?

**Comment by metasmile** [comment]
+1: any updates with some standard solution?

**Comment by wxindfx** [comment]
The same issue in Xcode 11.6. I found the temporary solution in my case. Run sourcery in terminal first, it will succeed. Then it will always succeed in XCode.

**Comment by krzysztofzablocki** [comment]
I think this might be issue with caching of Swift template binaries or some mismatch between what we bundle (SourceryFramework?), given it only happens with Swift templates (this has always been problematic), but @xploit42 said clearing cache didn't help so I'm not sure, I never really worked with SwiftTemplates, what do you think might be the issue @ilyapuchka @Liquidsoul ?

**Comment by csykes** [comment]
I don't have a simplified swift template file but here is how we got here. Everything was working fine using any version of Xcode 11.x with Sourcery 0.17 but when we upgraded to 0.18 we ran into a bunch of build errors along these lines. 

`Failed to unarchive /Users/xxxx/Library/Caches/Sourcery/ProjectName/-2737364290730679759.srf due to error, re-parsing`

Tried the usual Xcode project clean, deleting Xcode's derived data + cache but it made no difference. Deleting the Sourcery cache folder `~/Library/Caches/Sourcery` also had no effect other than making those unarchive/parsing errors to disappear. We then end up with the silent failure that most people refer to in this issue.

The scary thing is we reverted back to 0.17 and Xcode still failed so we'd effectively broken our development environment. Running Xcode's Sourcery build phase command on the command line did succeed and after that builds from Xcode started to work again as well.

**Comment by krzysztofzablocki** [comment]
@csykes the unarchive messages are totally expected as that's just cache re-invalidation after tool version change, I assume you use Swift template as well? sounds like a different version of toolchain is used when running via xcode build phase vs command line

**Comment by csykes** [comment]
Yes we are using a Swift template in our project but if I remove the contents from the template file the project still fails in the same way. So I don't think its related the contents of the swift template file and you are probably on the right path to think its related to a difference in the toolchain/caching mechanisms between Xcode vs CommandLine.

If it helps, the command to generate our files is based off the Sourcery Pod files we have stored in GitHub
`$PODS_ROOT/Sourcery/bin/sourcery --config PathToFile/.sourcery.yml`

And our .sourcery.yml
```
sources:
 - ./
templates:
 - GraphQL/SourceryTemplates/
output: ./
```

**Comment by Sega-Zero** [comment]
Any news on it?

**Comment by krzysztofzablocki** [comment]
@csykes is your command line tools path (`xcode-select -p`) same as your Xcode you use for building the project? I'm assuming that is why via console it works vs when using Xcode build phase

**Comment by csykes** [comment]
I believe the command line + Xcode are the same
<img width="400" alt="Screen Shot 2020-08-10 at 8 09 34 AM" src="https://user-images.githubusercontent.com/944809/89791706-dfc54b80-dae0-11ea-868f-f4bd2e21fdc1.png">

**Comment by DavidBertet** [comment]
I just opened a PR with the fix and an explanation of the problem if you are interested 👍

**Comment by krzysztofzablocki** [comment]
I've released Sourcery 1.0.0 (finally pulled the plug on proper version value 😂 ) with @DavidBertet fix, can you double check it solves your issue? I'll close the issue afterwards /cc @csykes @hartbit

**Comment by csykes** [comment]
Verified as working for me. Thanks everyone!
